backend/program/settings/models.py

```diff
@@ -0,0 +1,105 @@
+"""Iceberg settings models"""
+
+from typing import Optional
+from pydantic import BaseModel, root_validator
+
+
+
+class NotifyingBaseModel(BaseModel):
```

Comment:
this is so that we can auto-call the observers whenever a key is changed at any level of the settings